src/main/java/com/salesforce/cdp/queryservice/util/TokenHelper.java

```diff
@@ -95,6 +99,8 @@ private TokenHelper() {
             log.error("Caught exception while retrieving the token", e);
             invalidateCoreToken(properties.getProperty(Constants.LOGIN_URL), coreTokenRenewResponse == null ? null : coreTokenRenewResponse.getAccess_token(), client);
             throw new SQLException(TOKEN_EXCHANGE_FAILURE);
+        } finally {
+            Arrays.fill(passwordBytes, (byte)0);
```

Comment:
will this not change the properties map as well ? when we try to get the token again it will fail ?

Reply:
yes this changes property map. This will not fail as the fresh properties are sent from the client to JDBC driver with new connection. Tested this with DBeaver and the behaviour was fine.